src/NosCore.Controllers/Shops/SellPacketHandler.cs

```diff
@@ -42,66 +42,68 @@ public SellPacketHandler(WorldConfiguration worldConfiguration)
             _worldConfiguration = worldConfiguration;
         }
 
-        public override Task Execute(SellPacket sellPacket, ClientSession clientSession)
+        public override Task ExecuteAsync(SellPacket sellPacket, ClientSession clientSession)
         {
             if (clientSession.Character.InExchangeOrTrade)
             {
                 //TODO log
                 return Task.CompletedTask;
             }
 
-            if (sellPacket.Amount.HasValue && sellPacket.Slot.HasValue)
+            if (!sellPacket.Amount.HasValue || !sellPacket.Slot.HasValue)
             {
-                var type = (NoscorePocketType) sellPacket.Data;
-
-                var inv = clientSession.Character.InventoryService.LoadBySlotAndType(sellPacket.Slot.Value, type);
-                if ((inv == null) || (sellPacket.Amount.Value > inv.ItemInstance!.Amount))
-                {
-                    //TODO log
-                    return Task.CompletedTask;
-                }
+                return Task.CompletedTask;
+            }
 
-                if (!inv.ItemInstance.Item!.IsSoldable)
-                {
-                    clientSession.SendPacket(new SMemoPacket
-                    {
-                        Type = SMemoType.Error,
-                        Message = GameLanguage.Instance.GetMessageFromKey(LanguageKey.ITEM_NOT_SOLDABLE,
-                            clientSession.Account.Language)
-                    });
-                    return Task.CompletedTask;
-                }
+            var type = (NoscorePocketType) sellPacket.Data;
 
-                var price = inv.ItemInstance.Item.ItemType == ItemType.Sell ? inv.ItemInstance.Item.Price
-                    : inv.ItemInstance.Item.Price / 20;
+            var inv = clientSession.Character.InventoryService.LoadBySlotAndType(sellPacket.Slot.Value, type);
+            if ((inv == null) || (sellPacket.Amount.Value > inv.ItemInstance!.Amount))
+            {
+                //TODO log
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/manual/NosCoreIO/NosCore/pullRequest?prid=5266673)